gcsio/pom.xml

```diff
@@ -54,6 +54,24 @@
         </plugins>
       </build>
     </profile>
+    <profile>
+      <id>hadoop2</id>
+      <dependencies>
+        <dependency>
+          <groupId>org.apache.hadoop</groupId>
+          <artifactId>hadoop-common</artifactId>
+        </dependency>
+      </dependencies>
+    </profile>
+    <profile>
+      <id>hadoop3</id>
+      <dependencies>
+        <dependency>
+          <groupId>org.apache.hadoop</groupId>
+          <artifactId>hadoop-client-api</artifactId>
+        </dependency>
+      </dependencies>
+    </profile>
```

Comment:
GCSIO library can not depend on Hadoop classes, please remove.

Reply:
This is interesting. The module definition says `An implementation of org.apache.hadoop.fs.FileSystem targeting Google Cloud Storage`. I understand the separation of concerns, but It's unclear to me why we want to avoid depending on Hadoop classes when we're implementing `org.apache.hadoop.fs.FileSystem`.

Reply:
GCSIO does not implement `org.apache.hadoop.fs.FileSystem` it implements its own FS-like abstraction. GCSIO is Hadoop-free because it used by frameworks (Beam, etc) that do not depend on the Hadoop - this is a whole point of having `gcs/` and `gcsio/` modules - if both will depend on Hadoop we as well may just merge them together.

Reply:
Ok, this makes sense now. I will update the module definition (the statement above is copied from the pom.xml definition).